features/2025-07-radius-resource-types-contribution.md

```diff
@@ -0,0 +1,292 @@
+
+When I try to contribute a new resource type, I struggle to understand the different pathways available. Should I create a Radius resource type, contribute a Recipe, or modify the core Radius codebase?
+
+I lack clear guidance on how to structure my contribution. There's no template or example to follow, and I'm unsure about required documentation, testing procedures, or quality standards.
+
+I don't know how to test my resource type before submitting a PR.
+
+Also, how do I discover all the contributed resource types and Recipes?
+
+## Desired user experience outcome
+
+After the contribution experience is implemented, I can easily contribute resource types to the Radius community through a clear, well-documented process. I can choose from multiple contribution pathways based on my needs and technical expertise, with each pathway providing comprehensive guidance.
+
+I can validate my resource type and its associated Recipes before submitting it to the community.
+
+Once I contribute a resource type, it becomes discoverable in the repository, where other users can find, evaluate, and use it. 
+
+### Detailed user experience
+
+#### User story 1: As a platform engineer or an open-source contributor, I want to contribute a Radius Resource type so that I can share my work with the community and help others benefit from it.
+
+1. The contributor understands the contribution guidelines at the `resource-types-contrib` repository and familiarizes themselves with the [contribution process](https://github.com/reshrahim/resource-types-contrib/blob/main/CONTRIBUTING.md).
+
+1. They will pick the resource type that they want to contribute either from the list of open issues in the `resource-types-contrib` repository, or they can contribute their own resource type if it does not already exist.
+
+    > [!NOTE]
+    > We create a list of open issues in the `resource-types-contrib` repository that needs contribution starting with the portable Resource types and then move open requests for new resource types from `radius` repository to the `resource-types-contrib` repository.
+    >
+    >    | Namespace | Resource Type | Description |
+    >    |---|---|---|
+    >    | Radius.Datastores | sqlDatabases | Relational database support |
+    >    | Radius.Datastores | redisCaches | In-memory data store and cache |
+    >    | Radius.Datastores | mongoDatabases | NoSQL document database |
+    >    | Radius.Messaging | rabbitMQQueues | Message queue system for asynchronous communication |
+    >    | Radius.Dapr | stateStores | Distributed state management via Dapr |
+    >    | Radius.Dapr | pubSubBrokers | Message publishing and subscription via Dapr |
+    >    | Radius.Dapr | secretStores | Secret management via Dapr components |
+    >    | Radius.Dapr | configurationStores | Configuration management via Dapr |
+
+1. Create a new directory for your resource type under the `resource-types-contrib` repository, following the established directory structure.
```

Comment:
Can we add a visual pic from for the repo structure from the compute extensibility spec: https://github.com/willtsai/design-notes-radius/blob/compute-extensibility-feature-specs/architecture/2025-06-compute-extensibility-feature-spec/types-contrib-repo-structure.png